ksqldb-engine/src/main/java/io/confluent/ksql/query/QueryBuilder.java

```diff
@@ -235,6 +238,24 @@ TransientQueryMetadata buildTransientQuery(
     return Optional.empty();
   }
 
+  //This covers the queries that source-key format does not support.
+  private boolean isUnsupportedFormat(final KeyFormat pullQueryKeyFormat,
```

Comment:
I'm still really confused about this. As much as I "understand" the fix, we should be using the same serializer for looking up the key as we do when serializing it into the changelog topic. That should be true independently of whether or not the format uses schema registry or not.

Why is it different if the key is a struct or a decimal? Why is that different if the format is `KAFKA`?

I recall you saying that some QTTs failed without this method, we should figure out why that's the case because I'm worried we may be introducing a different bug. This shouldn't affect QTTs at all, right? It should only affect the pull query path.

At a minimum, we should add a detailed comment into the code explaining this (since it really is a "hack")

Reply:
Added comments and link of analysis to the description of PR.